lms/djangoapps/instructor/tasks.py

```diff
@@ -0,0 +1,96 @@
+    )
+    err_msg = None
+    try:
+        user = get_user_by_username_or_email(user_identifier)
+        block_key = UsageKey.from_string(content_id)
+        root_descriptor = modulestore().get_item(block_key)
+    except ObjectDoesNotExist:
+        err_msg = err_msg_prefix + 'User does not exist!'
+    except InvalidKeyError:
+        err_msg = err_msg_prefix + 'Invalid content_id!'
+    except ItemNotFoundError:
+        err_msg = err_msg_prefix + 'Block not found in the modulestore!'
+    if err_msg:
+        log.error(err_msg)
+        return
+
+    # This logic has been copied over from openedx/core/djangoapps/schedules/content_highlights.py
+    # in the _get_course_module function.
+    # I'm not sure if it's a pattern that should be copied so likely if it can be avoided, do that.
+    # We are using it here because we ran into issues with the User service being undefined when we
+    # encountered a split_test xblock.
+
+    # Fake a request to fool parts of the courseware that want to inspect it.
+    request = get_request_or_stub()
+    request.user = user
+
+    # Now evil modulestore magic to inflate our descriptor with user state and
+    # permissions checks.
+    field_data_cache = FieldDataCache.cache_for_descriptor_descendents(
+        root_descriptor.course_id, user, root_descriptor, read_only=True,
+    )
+    root_module = get_module_for_descriptor(
+        user, request, root_descriptor, field_data_cache, root_descriptor.course_id,
+    )
+    if not root_module:
+        err_msg = err_msg_prefix + 'Module unable to be created from descriptor!'
+        log.error(err_msg)
+        return
+
+    def _submit_completions(block, user):
+        """
+        Recursively submits the children for completion to the Completion Service
+        """
+        mode = XBlockCompletionMode.get_mode(block)
+        if mode == XBlockCompletionMode.COMPLETABLE:
+            block.runtime.publish(block, 'completion', {'completion': 1.0, 'user_id': user.id})
+        elif mode == XBlockCompletionMode.AGGREGATOR:
+            # I know this looks weird, but at the time of writing at least, there isn't a good
+            # single way to get the children assigned for a partcular user. Some blocks define the
+            # child descriptors method, but others don't and with blocks like Randomized Content
+            # (Library Content), the get_children method returns all children and not just assigned
+            # children. So this is our way around situations like that. See also Split Test Module
+            # for another use case where user state has to be taken into account via get_child_descriptors
+            block_children = ((hasattr(block, 'get_child_descriptors') and block.get_child_descriptors())
+                              or (hasattr(block, 'get_children') and block.get_children()))
+            for child in block_children:
+                _submit_completions(child, user)
```

Comment:
If a block has neither attribute, block_children will be False, yeah? Maybe an additional `or []` at the end of the block_children assignment line?